Fix line counting for large output display (issue #1973)

vicky-dx opened the pull request:
Improve console output confirmation by counting display lines more accurately when output is large.

Changes:
- Remove COUNT_LINES ifdef (keep line counting, drop char counting)
- Add count_display_lines() function
- Count long lines (>80 chars) as multiple display lines
- More accurate line count prevents misleading prompts like 'print 1 line?' for a 100MB single line

As suggested by @trufae, this makes the 'Do you want to print N lines?' prompt more meaningful by considering line wrapping.

Fixes #1973 (issue open since 2015)

trufae commented on libr/cons/cons.c:
```diff
@@ -13,6 +11,39 @@ static RCons s_cons_global = {0};
 
+// Count display lines, treating long lines (>80 chars) as multiple lines
+// Stops counting at R_CONS_MAX_LINES to avoid performance issues with huge outputs
+#define R_CONS_MAX_LINES 10000
```
let's use 9000, so we can make the joke and tell the user the lines are over 9000. also remove the R_CONS prefix here because its local to this file. R_CONS must be only the ones in the .h

trufae reviewed (changes requested):
almost there

trufae commented on libr/cons/cons.c:
```diff
@@ -13,6 +11,39 @@ static RCons s_cons_global = {0};
 
 static void __break_signal(int sig);
 
+// Count display lines, treating long lines (>80 chars) as multiple lines
```
remove comments

trufae commented on libr/cons/cons.c:
```diff
@@ -13,6 +11,39 @@ static RCons s_cons_global = {0};
+		} else {
+			line_chars++;
+			// Count extra line for every 80 characters in a single line
+			if (line_chars > 0 && line_chars % 80 == 0) {
```
let's remove the hardcoded "80" value here. use const int columns = r_cons_get_size

vicky-dx commented:
Hey @trufae , I've applied all your review feedback. Thank you for the detailed suggestions!

Changes made:

-  Changed R_CONS_MAX_LINES 10000 → MAX_DISPLAY_LINES 9000 🤣
-  Removed R_CONS_ prefix (local to file)
-  Removed unnecessary null/zero checks
-  Using r_cons_get_size(cons, NULL) to get terminal columns dynamically
-  Optimized with strchr() instead of char-by-char loop
-  Removed all comments
-  Fixed printf format from %d to %u

**Line wrapping calculation fix:**
I noticed the initial formula lines += 1 + (line_len / columns) had an off-by-one issue at boundaries:

- 80 chars was counting as 2 lines (should be 1)
- 160 chars was counting as 3 lines (should be 2)

**Fixed with ceiling division:**
```c
lines += line_len ? ((line_len + columns - 1) / columns) : 1;
```

Tested and verified:

- 80 chars = 1 line 
- 81 chars = 2 lines 
- 160 chars = 2 lines 
- 4000 chars = 50 lines
